// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package abesim;

import "agent_based_epidemic_sim/core/pandemic.proto";

message PTTSTransitionModelProto {
  message TransitionProbability {
    HealthState.State health_state = 1;
    float transition_probability = 2;
  }
  message TransitionProbabilities {
    HealthState.State health_state = 1;
    repeated TransitionProbability transition_probability = 2;
    // The rate at which an agent transitions to the next state.
    float rate = 3;
  }
  // The state transition diagram.
  repeated TransitionProbabilities state_transition_diagram = 1;
}
